browser/js/map/map.js

```diff
@@ -48,71 +48,113 @@ app.controller('MapCtrl', function($scope, MapFactory, FilterFactory, ReviewFact
-                $scope.currentMarkers[i].setMap(null);
-            }
+    // Adds all apartments to the map on initial page load
+    $scope.apartments.forEach(function(apartment) {
+        addMarkerToMap(apartment);
+    });
+
+    // Function to check all filter criteria
+    var checkAllCriteria = function(apartmentToCheck) {
+
+        // Helper functions to check each criteria
+        var checkBedrooms = function() {
+            if (!$scope.filterCriteria.numBedrooms) return true;
+            if ($scope.filterCriteria.numBedrooms.val === apartmentToCheck.numBedrooms) return true;
+            return false;
         }
+
+        var checkRent = function() {
+            var minRent = $scope.filterCriteria.monthlyPriceMin ? $scope.filterCriteria.monthlyPriceMin : 0;
+            var maxRent = $scope.filterCriteria.monthlyPriceMax ? $scope.filterCriteria.monthlyPriceMax : 1000000000;
+
+            if (apartmentToCheck.monthlyPrice >= minRent && apartmentToCheck.monthlyPrice <= maxRent) return true;
+            return false;
+        }
+
+        var checkTerm = function() {
+            if (!$scope.filterCriteria.termOfLease) return true;
+            if ($scope.filterCriteria.termOfLease === apartmentToCheck.termOfLease) return true;
+            return false;
+        }
+
+        // NOTE: Rating currently doesn't work
+        // var checkRating = function() {
+        //     if (!$scope.filterCriteria.averageRating) return true;
+        //     if (apartmentToCheck.averageRating >= $scope.filterCriteria.averageRating - 0.5) return true;
+        //     return false;
+        // }
+
+        if (!checkBedrooms()) return false;
+        if (!checkRent()) return false;
+        if (!checkTerm()) return false;
+        // if (!checkRating()) return false;
+
+        return true;
     }
 
+    // Filter results based on the criteria the users submits
     $scope.filterResults = function() {
-        FilterFactory.filterResults($scope.filterCriteria)
-            .then(function(apartments) {
-                console.log("Apartments found: ", apartments);
-                var filteredIds = apartments.map(function(apartment) {
-                    return apartment._id;
-                });
-                $scope.currentMarkers.forEach(function(marker) {
-                    if (filteredIds.indexOf(marker.apartmentId) === -1) marker.setMap(null);
-                    else marker.setMap($scope.map)
-                })
-            });
+        // Reset all of the map markers. Not sure if there's a better way to do this
+        $scope.currentMarkers.forEach(function(mapMarker) {
+            mapMarker.setMap(null);
+        });
+        $scope.currentMarkers = [];
+
+        // Loop over each apartment and check if a marker should be added
+        $scope.apartments.forEach(function(apartmentToCheck) {
+            if (checkAllCriteria(apartmentToCheck)) addMarkerToMap(apartmentToCheck);
+        });
+    }
```

Comment:
Instead of resetting the markers each time, maybe under line 64 you can add the properties that we are filtering by to createdMapMarker. Then within the forEach on line 131 you can do: 
if (checkAllCriteria(apartmentToCheck)) mapMarker.setMap(map)